R/defStormsDataset.R

```diff
@@ -241,23 +248,350 @@ checkInputsdefStormsDataset <- function(filename, fields, basin, seasons, unitCo
+    } else if (unitConversion["msw"] == "kmh2ms") {
+      msw[,i] <- as.numeric(c(kmh2ms(as.numeric(dataBaseFiltered[start:end, fields["msw"]])), rep(NaN, row-countObs[i])))
+    } else {
+      msw[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["msw"]], rep(NaN, row-countObs[i])))
+    }
+    
+    # TODO Remove later
+    if ("sshs" %in% names(fields)) {
+      sshs[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["sshs"]], rep(NaN, row-countObs[i])))
+    }
+    
+    if ("rmw" %in% names(fields)) {
+      if (unitConversion["rmw"] == "nm2km") {
+        rmw[,i] <- as.numeric(c(nm2km(as.numeric(dataBaseFiltered[start:end, fields["rmw"]])), rep(NaN, row-countObs[i])))
+      } else {
+        rmw[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["rwm"]], rep(NaN, row-countObs[i])))
+      }
+    }
+    
+    if ("pressure" %in% names(fields)) {
+      if (unitConversion["pressure"] == "mb2pa") {
+        pressure[,i] <- as.numeric(c(mb2pa(as.numeric(dataBaseFiltered[start:end, fields["pressure"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["pressure"] == "b2pa") {
+        pressure[,i] <- as.numeric(c(b2pa(as.numeric(dataBaseFiltered[start:end, fields["pressure"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["pressure"] == "psi2pa") {
+        pressure[,i] <- as.numeric(c(psi2pa(as.numeric(dataBaseFiltered[start:end, fields["pressure"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["pressure"] == "atm2pa") {
+        pressure[,i] <- as.numeric(c(atm2pa(as.numeric(dataBaseFiltered[start:end, fields["pressure"]])), rep(NaN, row-countObs[i])))
+
+      } else {
+        pressure[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["pressure"]], rep(NaN, row-countObs[i])))
+      }
+    }
+    
+    if ("poci" %in% names(fields)) {
+      if (unitConversion["poci"] == "mb2pa") {
+        poci[,i] <- as.numeric(c(mb2pa(as.numeric(dataBaseFiltered[start:end, fields["poci"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["poci"] == "b2pa") {
+        poci[,i] <- as.numeric(c(b2pa(as.numeric(dataBaseFiltered[start:end, fields["poci"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["poci"] == "psi2pa") {
+        poci[,i] <- as.numeric(c(psi2pa(as.numeric(dataBaseFiltered[start:end, fields["poci"]])), rep(NaN, row-countObs[i])))
+
+      } else if (unitConversion["poci"] == "atm2pa") {
+        poci[,i] <- as.numeric(c(atm2pa(as.numeric(dataBaseFiltered[start:end, fields["poci"]])), rep(NaN, row-countObs[i])))
+
+      } else {
+        poci[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["poci"]], rep(NaN, row-countObs[i])))
+      }
+    }
+
+  }
+  
+  # Collect data
+  data <- list(
+    names = names,
+    seasons = seasons,
+    isotimes = isotimes,
+    longitude = longitude,
+    latitude = latitude,
+    msw = msw
+  )
+  
+  # TODO Remove later
```

Comment:
Do you plan to directly deal with the data object rather intermediate variables and re-affect the values at the end ?

Reply:
Done